packages/components/src/modal/index.tsx

```diff
@@ -39,6 +39,18 @@ import type { ModalProps } from './types';
 // Used to count the number of open modals.
 let openModalCount = 0;
 
+/**
+ * Returns the first tabbable element that is not a close button.
+ * See: https://github.com/WordPress/gutenberg/issues/54106.
+ * @param tabbables HTMLElement[] an array of tabbable elements.
+ * @return HTMLElement the first tabbable element that is not a close button.
+ */
+function focusFirstNonCloseButtonElement( tabbables: HTMLElement[] ) {
+	return tabbables.find(
+		( tabbableNode ) => tabbableNode?.ariaLabel !== 'Close'
+	);
```

Comment:
One issue I can see with this approach is internationalisation – it won't generally work in non-English environments. Additionally, "Close" is an overridable value (see `closeButtonLabel`), so there's no guarantee it'll be that in English either.

Linguistics aside though, developers can provide additional `headerActions`, which I'd posit should probably also be skipped for the purposes of this exercise. So rather than explicitly looking for a node labelled 'Close', it might be better to find the first relevant node inside `childrenContainerRef`, which wraps the content of the dialog.

Reply:
Nice. Great points. I'll look into this. Thanks!

Reply:
Both points should now be addressed by way of updates that now optimise for ignoring the entire header area.